> apk, macports and dpkgpng all use their own action "fetch" when "-w" (download-only) option is specified. Because the method is invoked with $_EOPT as part of the argument (in our case, $_EOPT == fetch), we need to remove the part before processing the remained parts.
> 
> Fix #214

This fỉrst attempt in this PR doesn't actually get rid of the issue. There is a potential case. Let's look

```
_translate_all:
  export _EOPT="${local_args# }"
```

`_EOPT` can be more than a single term `fetch`. Recall how `_EOPT` is used at the very end of the `zz_main.sh` script:

```
zz_main.sh:
  "_${_PACMAN}_init" || exit
  # shellcheck disable=SC2086
  "${_PACMAN}_${_POPT}${_SOPT}" $_EOPT "$@"
```

What happens if more options are provided. Luckily it's there by accident (:D):

```
_translate_all:
  local_args="$(_translate_w)" || return 1
  local_args="${local_args}${local_noconfirm:+ }${local_noconfirm}"
  local_args="${local_args}${local_debug:+ }${local_debug}"
```
so, `fetch` would be the very first thing in all situations!

But in case of `apk`, we can have a problem:

```
root@be744a7db20a:/src# PACAPT_DEBUG=apk ./pacapt.dev -S --noconfirm -w -v nano
:: [debug] Switching to Bash shell
pacapt: apk, p=S, s=, t=, e=fetch -v
pacapt: execute 'apk_S fetch -v nano'
apk_S () 
{ 
    case ${_EOPT} in 
        "fetch")
            shift;
            apk fetch $_TOPT -- "$@"
        ;;
        *)
            apk add $_TOPT -- "$@"
        ;;
    esac
}
```

The two dashes `--` can be wrong. So I'm going to remove that too.

--- a/lib/00_core.sh
+++ b/lib/00_core.sh
@@ -237,8 +237,10 @@ _translate_all() {
 
   local_debug="$(_translate_debug)" || return 1
   local_noconfirm="$(_translate_noconfirm)" || return 1
-  local_args="$(_translate_w)" || return 1
 
+  # WARNING: Order does matter, see also
+  #   https://github.com/icy/pacapt/pull/219#issuecomment-1006079629
+  local_args="$(_translate_w)" || return 1
   local_args="${local_args}${local_noconfirm:+ }${local_noconfirm}"
   local_args="${local_args}${local_debug:+ }${local_debug}"
 

--- a/lib/apk.sh
+++ b/lib/apk.sh
@@ -116,8 +116,8 @@ apk_S() {
   case ${_EOPT} in
     # Download only
     ("fetch") shift
-              apk fetch $_TOPT -- "$@" ;;
-          (*) apk add   $_TOPT -- "$@" ;;
+              apk fetch $_TOPT "$@" ;;
+          (*) apk add   $_TOPT "$@" ;;
   esac
 }
 

